pre-commit.sh

```diff
@@ -1,12 +1,26 @@
 #!/usr/bin/env bash
 
-if [ "$1" == "--no-browser" ]
-then
-    SHOW_BROWSER=false
-fi
+while [[ $# -gt 0 ]]
+do
+    key="$1"
+    case $key in
```

Comment:
could this have some explanation and clarity what all of this is doing? and ideally as a separate PR maybe to say this is only for future use and not specific for this ticket?